File: python/binary_tree_traverse.py
import sys
import time

class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def build_tree(depth, counter):
    if depth <= 0:
        return None
    
    node = Node(counter[0])
    counter[0] += 1
    node.left = build_tree(depth - 1, counter)
    node.right = build_tree(depth - 1, counter)
    return node

def traverse_in_order(node):
    if node is None:
        return 0
    
    sum_val = 0
    sum_val += traverse_in_order(node.left)
    sum_val += node.data
    sum_val += traverse_in_order(node.right)
    return sum_val

if len(sys.argv) < 2:
    print('Usage: binary_tree_traverse <depth>')
    sys.exit(1)

try:
    depth = int(sys.argv[1])
except ValueError:
    print('Invalid number:', sys.argv[1])
    sys.exit(1)

counter = [0]

start = time.time()
root = build_tree(depth, counter)
sum_val = traverse_in_order(root)
end = time.time()

time_ms = int((end - start) * 1000)

print(f'Python: binary_tree_traverse({depth}) = {sum_val}')
print(f'Time: {time_ms}ms')

File: swift/binary_tree_traverse.swift
import Foundation

class Node {
    var data: Int
    var left: Node?
    var right: Node?
    
    init(_ data: Int) {
        self.data = data
        self.left = nil
        self.right = nil
    }
}

func buildTree(_ depth: Int, _ counter: inout Int) -> Node? {
    if depth <= 0 { return nil }
    
    let node = Node(counter)
    counter += 1
    node.left = buildTree(depth - 1, &counter)
    node.right = buildTree(depth - 1, &counter)
    return node
}

func traverseInOrder(_ node: Node?) -> Int {
    guard let node = node else { return 0 }
    
    var sum = 0
    sum += traverseInOrder(node.left)
    sum += node.data
    sum += traverseInOrder(node.right)
    return sum
}

let args = CommandLine.arguments
if args.count < 2 {
    print("Usage: binary_tree_traverse <depth>")
    exit(1)
}

guard let depth = Int(args[1]) else {
    print("Invalid number: \(args[1])")
    exit(1)
}

var counter = 0

let start = CFAbsoluteTimeGetCurrent()
let root = buildTree(depth, &counter)
let sum = traverseInOrder(root)
let end = CFAbsoluteTimeGetCurrent()

let timeMs = Int((end - start) * 1000)

print("Swift: binary_tree_traverse(\(depth)) = \(sum)")
print("Time: \(timeMs)ms")